docs/man/rpm-dependency-generators.7.scd

```diff
@@ -0,0 +1,387 @@
+*%\_\_*_NAME_*_exclude_magic* _REGEX_++
+*%\_\_*_NAME_*_exclude_mime* _REGEX_
+
+## Flags
+
+The following macros allow for tweaking some aspects of the inclusion and
+exclusion matching logic, respectively:
+
+*%\_\_*_NAME_*_flags* _FLAGS_++
+*%\_\_*_NAME_*_exclude_flags* _FLAGS_
+
+_FLAGS_ is a comma-separated list of the following possible values:
+
+- *exeonly*: require executable bit set
+- *magic_and_path*: require both magic and path to match
+
+# GENERATORS
+
+A generator is an executable that reads filenames from stdin and writes
+dependencies of a single _type_ to stdout, according to a _protocol_. It is
+written for a specific file attribute and defined as a macro in the respective
+_NAME.attr_ file. Zero or more generators, one for each supported type, can be
+defined for a single file attribute.
+
+## Types
+
+The following generator macros are currently supported:
+
+*%\_\_*_NAME_*_conflicts* _BODY_++
+*%\_\_*_NAME_*_enhances* _BODY_++
+*%\_\_*_NAME_*_obsoletes* _BODY_++
+*%\_\_*_NAME_*_orderwithrequires* _BODY_ (Added: 4.17.0)++
+*%\_\_*_NAME_*_provides* _BODY_++
+*%\_\_*_NAME_*_recommends* _BODY_++
+*%\_\_*_NAME_*_requires* _BODY_++
+*%\_\_*_NAME_*_suggests* _BODY_++
+*%\_\_*_NAME_*_supplements* _BODY_
+
+_BODY_ is a command line (the program name and any arguments) that should be
+executed for every file that has the attribute _NAME_, and is macro-expanded on
+use.
+
+## Protocols
+
+The following generator protocols are currently supported:
+
+*simple*
+	Receives one filename at a time, outputs dependencies, one per line.
+
+*multifile*
+	Receives all filenames having the file attribute, one per line, and
+	outputs dependencies, one per line. The processed filename prepended
+	with *;* must be printed before printing dependencies for that file.
+
+	This is the recommended protocol for newly written generators.
+
+	Added: 4.20.0
+
+By default, the *simple* protocol is used. To enable the *multifile* protocol
+for all generators of the file attribute _NAME_, define the following macro:
+
+	*%\_\_*_NAME_*_protocol multifile*
+
+Unlike the path _REGEX_ in file attributes, generators receive filenames with
+the *%{buildroot}* prefix so that they can access the actual file contents on
+disk.
+
+Due to backwards compatibility, generators must always consume all of stdin and
+not make any assumptions about the number of files passed, regardless of the
+protocol used.
+
+## Exported macros
+
+In addition to globally defined macros, the following macros are automatically
+exported to generators, on a subpackage basis:
+
+*%{name}*++
+*%{epoch}*++
+*%{version}*++
+*%{release}*
+
+Added: 4.15.0
+
+## Parametric macros
+
+If a generator is declared as a parametric macro, the macro itself is called for
+dependency generation, with the filename as its first argument (*%1*). The macro
+expansion itself is considered as the generated dependencies, one per line.
+
+This allows for writing efficient and powerful generators, especially when
```

Comment:
This sentence sounds like a sales pitch :smile: We need to communicate the reasons why one would want to use parametric generators over the other options, but I'd ditch at least the "powerful" part.

Reply:
Lol, indeed. I'll move to the original piece of text here, which was:
While RPM's own macro primitives are limited, using %{lua} enables the writing of complicated, macro-only generators.